gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/PropertyValueStrategyFactory.java

```diff
@@ -0,0 +1,220 @@
+   */
+  private final NullStrategy nullStrategy = new NullStrategy();
+
+  /**
+   * Constructs an {@code PropertyValueStrategyFactory} with type - strategy mappings as defined in
+   * {@code initClassStrategyMap}.
+   * Only one instance of this class is needed.
+   */
+  private PropertyValueStrategyFactory() {
+    classStrategyMap = initClassStrategyMap();
+    byteStrategyMap = initByteStrategyMap();
+  }
+
+  /**
+   * Returns value which is represented by the the provided byte array. Assumes that the value is
+   * serialized according to the {@code PropertyValue} standard.
+   *
+   * @param bytes byte array of raw bytes.
+   * @return Object which is the result of the deserialization.
+   */
+  public static Object fromRawBytes(byte[] bytes) {
+    PropertyValueStrategy strategy = INSTANCE.byteStrategyMap.get(bytes[0]);
+    return strategy == null ? null : strategy.get(bytes);
+  }
+
+  /**
+   * Compares two values.
+   * If {@code other} is not comparable to {@code value}, the used {@link PropertyValueStrategy}
+   * will throw an {@code IllegalArgumentException}.
+   *
+   * @param value first value.
+   * @param other second value.
+   * @return a negative integer, zero, or a positive integer as {@code value} is less than, equal
+   * to, or greater than {@code other}.
+   */
+  public static int compare(Object value, Object other) {
+    if (value == null) {
+      return INSTANCE.nullStrategy.compare(null, other);
+    } else {
+      PropertyValueStrategy strategy = get(value.getClass());
+      return strategy.compare(value, other);
+    }
+  }
+
+  /**
+   * Get byte array representation of the provided object. The object is serialized according to the
+   * {@code PropertyValue} standard.
+   * If the given type is not supported, an {@code UnsupportedTypeException} will be thrown.
+   *
+   * @param value to be serialized.
+   * @return byte array representation of the provided object.
+   */
+  public static byte[] getRawBytes(Object value) {
+    if (value != null) {
+      return get(value.getClass()).getRawBytes(value);
+    }
+    return new byte[] {PropertyValue.TYPE_NULL};
+  }
+
+  /**
+   * Returns strategy mapping to the provided byte value.
+   *
+   * @param value representing a data type.
+   * @return strategy class.
+   * @throws UnsupportedTypeException when there is no matching strategy for a given type byte.
+   */
+  public static PropertyValueStrategy get(byte value) throws UnsupportedTypeException {
+    PropertyValueStrategy strategy = INSTANCE.byteStrategyMap.get(value);
+    if (strategy == null) {
+      throw new UnsupportedTypeException("No strategy for type byte " + value);
+    }
+
+    return strategy;
+  }
+
+  /**
+   * Get a strategy which corresponds the provided class. If there is no mapping for the provided
+   * class in the class-strategy map, or the value of the parameter is {@code null}, an instance of
+   * {@code NoopPropertyValue} is returned.
```

Comment:
Does NoopPropertyValue still exist?

Reply:
NoopPopertyValue has been renamed to NullStrategy. Changed comment accordingly